iphone/Classes/KrollContext.m

```diff
@@ -1529,3 +1544,84 @@ - (void)dealloc
 }
 
 @end
+
+#if defined(USE_JSCORE_FRAMEWORK) && !defined(TI_USE_KROLL_THREAD)
+
+@implementation JSTimerManager
+
+- (instancetype)initInContext:(JSContext *)context
+{
+  self = [super init];
+  if (!self) {
+    return nil;
+  }
+
+  self.timers = [NSMutableDictionary new];
+
+  int (^setInterval)(void) = ^() {
+    return [self setIntervalFromArguments:JSContext.currentArguments shouldRepeat:YES];
+  };
+  context[@"setInterval"] = setInterval;
+
+  int (^setTimeout)(void) = ^() {
+    return [self setIntervalFromArguments:JSContext.currentArguments shouldRepeat:NO];
+  };
+  context[@"setTimeout"] = setTimeout;
+
+  void (^clearInterval)(JSValue *) = ^(JSValue *value) {
+    return [self clearIntervalWithIdentifier:value.toInt32];
+  };
+  context[@"clearInterval"] = clearInterval;
+  context[@"clearTimeout"] = clearInterval;
+
+  return self;
+}
+
+- (void)dealloc
+{
+  if (self.timers != nil) {
+    [self.timers removeAllObjects];
+    [self.timers release];
+    self.timers = nil;
+  }
+
+  [super dealloc];
+}
+
+- (void)invalidateAllTimers
+{
+  for (NSNumber *timerIdentifier in self.timers.allKeys) {
+    [self clearIntervalWithIdentifier:timerIdentifier.intValue];
+  }
+}
+
+- (int)setIntervalFromArguments:(NSArray<JSValue *> *)arguments shouldRepeat:(BOOL)shouldRepeat
+{
+  NSMutableArray *callbackArgs = [arguments.mutableCopy autorelease];
+  JSValue *callbackFunction = [callbackArgs objectAtIndex:0];
+  [callbackArgs removeObjectAtIndex:0];
+  double interval = [[callbackArgs objectAtIndex:0] toDouble] / 1000;
+  [callbackArgs removeObjectAtIndex:0];
+  NSNumber *timerIdentifier = [NSNumber numberWithUnsignedInteger:[[[[NSUUID alloc] init] autorelease] hash]];
+  NSTimer *timer = [NSTimer scheduledTimerWithTimeInterval:interval
+                                                   repeats:shouldRepeat
+                                                     block:^(NSTimer *_Nonnull timer) {
+                                                       [callbackFunction callWithArguments:callbackArgs];
+                                                       if (!shouldRepeat) {
+                                                         [self.timers removeObjectForKey:timerIdentifier];
+                                                       }
+                                                     }];
+  self.timers[timerIdentifier] = timer;
+  return [timerIdentifier intValue];
+}
+
+- (void)clearIntervalWithIdentifier:(int)identifier
+{
+  NSNumber *timerIdentifier = [NSNumber numberWithInteger:identifier];
```

Comment:
In general: All `[NSNumber numberWith****]` convenience constructors should use the boxed version of it (`@(identifier)`) since Objective C 2.0.